messageBus/index.js

```diff
@@ -0,0 +1,44 @@
+const sqs = require('./amazonSQS');
+
+const TOPIC_SEARCH = 'search';
+
+const publishSearchEvent = (searchEventId, params, results, timeline) => {
+  const {
+    visitId, userId, market, checkin, checkout, roomType, limit,
+  } = params;
+
+  const messagePayload = {
+    searchEventId,
+    request: {
+      timestamp: timeline.start.timestamp,
+      visitId,
+      userId,
+      market,
+      roomType: roomType || 'any',
+      limit: limit || 'no limit',
+    },
+    results,
+    timeline,
+  };
+  if (checkin && checkout) {
+    messagePayload.request.checkIn = checkin;
+    messagePayload.request.checkOut = checkout;
+  }
+
+  sqs.publish({ topic: TOPIC_SEARCH, payload: messagePayload });
+};
+
+const checkForMessages = topic =>
+  sqs.poll()
+    .then((messages) => {
+      const relevantMessages = messages.filter(message => JSON.parse(message.Body).topic === topic);
```

Comment:
You should probably grab messages by topic, rather than grabbing them all, and then filtering it by topic.

Reply:
Yes, that would be ideal but I was not able to figure out how to do this. If you know how to do this, please let me know!